File: 74-search-a-2d-matrix/74-search-a-2d-matrix.py
class Solution(object):
    def searchMatrix(self, matrix, target):
        """
        :type matrix: List[List[int]]
        :type target: int
        :rtype: bool
        """
        row_len = len(matrix)
        col_len = len(matrix[0])
    

        start = 0
        end = row_len*col_len-1
        while(start<=end):
            mid = (start+end)//2
            r = mid//col_len
            c = mid%col_len
            if(matrix[r][c]==target):
                return True
            elif(matrix[r][c]>target):
                end = mid -1
            else:
                start = mid+1
        return False

File: 74-search-a-2d-matrix/74-search-a-2d-matrix.js
/**
 * @param {number[][]} matrix
 * @param {number} target
 * @return {boolean}
 */
var searchMatrix = function(matrix, target) {
    let r_len = matrix.length;
    let c_len = matrix[0].length;
    
    let start = 0;
    let end = r_len*c_len -1;
    
    while(start <= end){
        let mid = Math.floor((start+end)/2);
        let r = Math.floor(mid/c_len) ;
        let c = mid % c_len;
        if(matrix[r][c]==target){
            return true
        }
        else if(matrix[r][c]>target){
            end = mid -1
        }
        else{
            start = mid +1
        }
    }
    return false
};